nilearn/plotting/html_stat_map.py

```diff
@@ -1,148 +1,438 @@
+    vsize = np.min(np.abs(s))
+    img = resample_img(img, target_affine=np.diag([vsize, vsize, vsize]),
+                       interpolation=interpolation)
+    return img
+
+
+def _data2sprite(data):
+    """ For internal use.
+        Convert a 3D array into a sprite of sagittal slices.
+    """
+
+    nx, ny, nz = data.shape
+    nrows = int(np.ceil(np.sqrt(nx)))
+    ncolumns = int(np.ceil(nx / float(nrows)))
+
+    sprite = np.zeros((nrows * nz, ncolumns * ny))
+    indrow, indcol = np.where(np.ones((nrows, ncolumns)))
+
+    for xx in range(nx):
+        # we need to flip the image in the x axis
+        sprite[(indrow[xx] * nz):((indrow[xx] + 1) * nz), (indcol[xx] * ny):
+               ((indcol[xx] + 1) * ny)] = data[xx, :, ::-1].transpose()
+
+    return sprite
+
+
+def _get_vmin_vmax(data, vmin=None, vmax=None):
+    """ For internal use.
+        Detect vmin and vmax in a data array.
+    """
+
+    # Get vmin vmax
+    show_nan_msg = False
+    if vmax is not None and np.isnan(vmax):
+        vmax = None
+        show_nan_msg = True
+    if vmin is not None and np.isnan(vmin):
+        vmin = None
+        show_nan_msg = True
+    if show_nan_msg:
+        nan_msg = ('NaN is not permitted for the vmax and vmin arguments.\n'
+                   'Tip: Use np.nanmax() instead of np.max().')
+        warnings.warn(nan_msg)
+
+    if vmax is None:
+        vmax = np.nanmax(data)
+    if vmin is None:
+        vmin = np.nanmin(data)
+    return vmin, vmax
+
+
+def _threshold_data(data, threshold=None):
+    """ For internal use only.
+        Threshold a data array.
+    """
+    # Deal with automatic settings of plot parameters
+    if threshold == 'auto':
+        # Threshold epsilon below a percentile value, to be sure that some
+        # voxels pass the threshold
+        threshold = fast_abs_percentile(data) - 1e-5
+
+    # Threshold
+    threshold = float(threshold) if threshold is not None else None
+
+    # Mask data
+    if threshold is not None:
+        if threshold == 0:
+            data = np.ma.masked_equal(data, 0, copy=False)
+        else:
+            data = np.ma.masked_inside(data, -threshold, threshold, copy=False)
+    return data
+
+
+def _save_sprite(data, output_sprite, vmax, vmin, mask=None, cmap='Greys',
+                 format='png'):
+    """ For internal use.
+        Generate a sprite from a 3D Niimg-like object.
+    """
+
+    # Create sprite
+    sprite = _data2sprite(data)
+
+    # Mask the sprite
+    if mask is not None:
+        mask = _data2sprite(mask)
+        sprite = np.ma.array(sprite, mask=mask)
+
+    # Save the sprite
+    imsave(output_sprite, sprite, vmin=vmin, vmax=vmax, cmap=cmap,
+           format=format)
+
+    return sprite
+
+
+def _bytesIO_to_base64(file):
+    """ For internal use.
+        Encode a bytesIO virtual file as base64.
+        Also closes the file.
+    """
+    file.seek(0)
+    data = encodebytes(file.read()).decode('utf-8')
+    file.close()
+    return data
```

Comment:
I take back my inlining comment, give that it is used in several places of the codebase.

Reply:
by the way, file is a keyword in python 2 so it would be better to choose a different name

Reply:
A good way to name the variable would be to indicate the use of the file (for example, sprite_file, if that is the info the variable will have).

Reply:
so I decided to stay consistent with the name of the function, and went for `bytesio_png`, because the input is a bytesio virtual file, storing a png image. It's used twice, once for the colormap, and once for the sprite. I guess that code could work with other things as well, but as long as it lives in html_stat_map.py this should be fine.